runtime/lua/vim/lsp/protocol.lua

```diff
@@ -923,4 +923,330 @@ function protocol.resolve_capabilities(server_capabilities)
+  --A request to provide folding ranges in a document. The request's
+  --parameter is of type {@link FoldingRangeParams}, the
+  --response is of type {@link FoldingRangeList} or a Thenable
+  --that resolves to such.
+  --@param FoldingRangeParams
+  FoldingRange = 'textDocument/foldingRange',
+  --A request to resolve the type definition locations of a symbol at a given text
+  --document position. The request's parameter is of type [TextDocumentPositionParams]
+  --(#TextDocumentPositionParams) the response is of type {@link Declaration}
+  --or a typed array of {@link DeclarationLink} or a Thenable that resolves
+  --to such.
+  --@param DeclarationParams
+  Declaration = 'textDocument/declaration',
+  --A request to provide selection ranges in a document. The request's
+  --parameter is of type {@link SelectionRangeParams}, the
+  --response is of type {@link SelectionRange SelectionRange[]} or a Thenable
+  --that resolves to such.
+  --@param SelectionRangeParams
+  SelectionRange = 'textDocument/selectionRange',
+  --The `window/workDoneProgress/create` request is sent from the server to the client to initiate progress
+  --reporting from the server.
+  --@param WorkDoneProgressCreateParams
+  WindowWorkDoneProgressCreate = 'window/workDoneProgress/create',
+  --A request to result a `CallHierarchyItem` in a document at a given position.
+  --Can be used as an input to an incoming or outgoing call hierarchy.
+  --@since 3.16.0
+  --@param CallHierarchyPrepareParams
+  PrepareCallHierarchy = 'textDocument/prepareCallHierarchy',
+  --A request to resolve the incoming calls for a given `CallHierarchyItem`.
+  --@since 3.16.0
+  --@param CallHierarchyIncomingCallsParams
+  CallHierarchyIncomingCalls = 'callHierarchy/incomingCalls',
+  --A request to resolve the outgoing calls for a given `CallHierarchyItem`.
+  --@since 3.16.0
+  --@param CallHierarchyOutgoingCallsParams
+  CallHierarchyOutgoingCalls = 'callHierarchy/outgoingCalls',
+  --@since 3.16.0
+  --@param SemanticTokensParams
+  SemanticTokensFull = 'textDocument/semanticTokens/full',
+  --@since 3.16.0
+  --@param SemanticTokensDeltaParams
+  SemanticTokensFullDelta = 'textDocument/semanticTokens/full/delta',
+  --@since 3.16.0
+  --@param SemanticTokensRangeParams
+  SemanticTokensRange = 'textDocument/semanticTokens/range',
+  --@since 3.16.0
+  WorkspaceSemanticTokensRefresh = 'workspace/semanticTokens/refresh',
+  --A request to show a document. This request might open an
+  --external program depending on the value of the URI to open.
+  --For example a request to open `https://code.visualstudio.com/`
+  --will very likely open the URI in a WEB browser.
+  --@since 3.16.0
+  --@param ShowDocumentParams
+  WindowShowDocument = 'window/showDocument',
+  --A request to provide ranges that can be edited together.
+  --@since 3.16.0
+  --@param LinkedEditingRangeParams
+  LinkedEditingRange = 'textDocument/linkedEditingRange',
```

Comment:
why do the `workspace/` symbols have Workspace prefix, but the `textDocument/` symbols don't? I asked this above...

Random quirky differences generally things harder to understand and require more documentation. So that should be avoided.


```suggestion
  textDocument_linkedEditingRange = 'textDocument/linkedEditingRange',
```

Reply:
Alternatively:

```suggestion
  textDocument = {
    linkedEditingRange = 'textDocument/linkedEditingRange',
  }
```

Reply:
Some common language features usually start with the textDocument prefix. The workspace has some special features (in order to use the workspace of vscode, they designed it) some language feature requests will contain some subsequent workspace requests.

Reply:
The question is about consistency (lack thereof).

Reply:
> Some common language features usually start with the textDocument prefix.

These are low-level symbols. We don't need to take shortcuts or make things clever. Let's be extra consistent here.

Reply:
this way might have problem.

```lua
 textDocument = {
    linkedEditingRange = 'textDocument/linkedEditingRange',
    foo = 'textDocument/foo',
    ...
  }
```
 some methods have  prefix some methods not . but they are both work for `inlayhint`..so which way ? convert `/` to `_`
```
textDocument/inlayHint
inlayHint/resolve
workspace/inlayHint/refresh
```

Reply:
If you allow me a vote here, I prefer something like `textDocument_linkedEditingRange = 'textDocument/linkedEditingRange'` than `textDocument = { linkedEditingRange = 'textDocument/linkedEditingRange' }`, since the former would result in code like:
```lua
local methods = vim.lsp.protocol.Methods

-- From just the line below, it is clear which method we're referring to.
if client.supports_method methods.textDocument_linkedEditingRange:
   ...
```

while the latter could result in code like:
```lua
local methods = vim.lsp.protocol.Methods.textDocument

-- Harder to understand because of the differences with the names used by the LSP specs.
if client.supports_method methods.linkedEditingRange:
```

Of course this is up to the person using these new constants, but still.